lib/default.nix

```diff
@@ -73,8 +73,12 @@ in {
       comps = config.components;
       applyLibrary = cname: f { cname = config.package.identifier.name; ctype = "lib"; };
       applySubComp = ctype: cname: f { inherit cname; ctype = componentPrefix.${ctype} or (throw "Missing component mapping for ${ctype}."); };
-      buildableAttrs = lib.filterAttrs (n: comp: config.package.buildable or true && comp.buildable or true && comp.planned);
-      libComp = if comps.library == null || !(config.package.buildable or true && comps.library.buildable or true && comps.library.planned)
+      isBuildable = comp:
```

Comment:
can we stick this down around L89 with the other "virtual properties" on things? I think users of haskell.nix might want this function too

Reply:
`config` would be out of scope there.  We could add it as an arg, but then I would worry that people would pass the wrong `config`.

Reply:
That seems like a potential pitfall with this approach then: now you can't tell if a component is buildable given just the component :grimacing:

Reply:
This `applyComponents` function filters out the ones that are not `buildable`, so if you have a component it is most likely buildable (otherwise you somehow bypassed `applyComponents`).

If you know a component might exist and want to check if it is `buildable` you could probably use something like `hsPkgs.packagename.components ? library` or `hsPkgs.packageName.components.exes ? exeName`).  Not ideal since the component could also have been deleted from `.cabal` file or something.

Reply:
I guess it's relatively unlikely that someone is going to be looking at these flags for other reasons.